### PR TITLE
Unit test for metric stream duplication

### PR DESCRIPTION
**Pull request by cijothomas**
Fixes #.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs [hunk 1]
```diff
@@ -38,6 +38,60 @@ public MetricApiTest(ITestOutputHelper output)
...
         }
 
+        [Theory]
+        [InlineData(AggregationTemporality.Cumulative)]
+        [InlineData(AggregationTemporality.Delta)]
+        public void StreamNamesDuplicatesAreNotAllowedTest(AggregationTemporality temporality)
+        {
+            var metricItems = new List<Metric>();
+            int metricCount = 0;
+            var metricExporter = new TestExporter<Metric>(ProcessExport);
```
Comment (reyang) [inline test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs:48]: Could we use InMemoryExporter?

Main reason is that we build InMemoryExporter for folks to use in their unit tests, I think we need to use it in our own tests, too.

Comment (cijothomas) [reply test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs:48]: Yes. Once https://github.com/open-telemetry/opentelemetry-dotnet/issues/2361 is fixed.

Comment (reyang) [reply test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs:48]: Does #2361 block this change?

Comment (cijothomas) [reply test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs:48]: no. this particular test is not blocked. 
Want to switch all the test in this class to use InMemoryExporter together once #2361 is ready.

**Review by reyang — APPROVED** [review]

**Pull request merged**